src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs

```diff
@@ -1608,6 +1721,121 @@ public bool Equals(Program other)
 parameters: CSharp6);
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateEqualsAndGetHashCode)]
+        public async Task TestImplementIEquatableOnClassInNullableContextWithUnannotatedMetadata()
+        {
+            await TestWithPickMembersDialogAsync(
+@"#nullable enable
+
+class Foo
+{
+    public int Bar { get; }
+    [||]
+}",
+@"#nullable enable
+
+using System;
+
+class Foo : IEquatable<Foo>
```

Comment:
This should be `IEquatable<Foo?>`

Reply:
I'm pretty sure I don't agree, but I might not be the best person to talk about this. I think it should be `IEquatable<this>` where `this` is precisely the same type as the implementing type.

Keep in mind that the official definition is `IEquatable<T>.Equals([AllowNull] T)` and when `T` is substituted with a reference type `Foo`, the official definition is:

`IEquatable<Foo>.Equals(Foo?)`

Reply:
If `Foo` was a struct, it wouldn't implement `IEquatable<Foo?>`. It seems like the Equals API was specifically designed so that reference and value types could remain consistent in that they implement IEquatable of their exact type.

Reply:
@sharwell Since this PR doesn't regress the thing you're talking about, if indeed it will end up getting changed, can that be done in a separate PR so the discussion can be decoupled and happen in a new issue?

Reply:
Since `Foo` is a reference type, `IEquatable<Foo?>` is the more accurate representation here. This change is required as part of updating this refactoring to support nullable reference types.

Note that an instance of `IEquatable<Foo?>` can be assigned to something requiring `IEquatable<Foo>`, but not the reverse.

Reply:
I can't see how it's more accurate since the `Equals` method ends up with the same `Foo?` parameter type either way.

> Note that an instance of `IEquatable<Foo?>` can be assigned to something requiring `IEquatable<Foo>`, but not the reverse.

This is a great point. I like this, will update.